blog/_posts/2020-05-26-create-your-own-theme-detailed-walkthrough.md

```diff
@@ -0,0 +1,372 @@
+---
+layout: post
+title: Create your own theme in minutes !
+---
+
+In Convos [v4.xx](https://convos.chat/blog/2020/5/14/theming-support-in-4-point-oh.html) it is super easy to customize
+or create a theme.  Follow this detailed walkthrough where I create a **"hack3r"** theme ! 
+
+<!--more-->
+
+
+## Smooth start
+
+You have to connect to your server where convos is running or you can do this tutorial locally : 
+
+`cd $CONVOS_HOME/public/themes`
+
+Then copy past an existing theme :
+
+`cp high-contrast_color-scheme-light.css mytheme.css`
+
+The name of your file is not important, the theme name will be included in the header of the CSS file.
+
+`vi mytheme.css`
+
+Choose your theme name :
+
+``` /*
+ * name: Hacker */ ```
+
+When you go to your convos instance settings, you will already be able to select your new theme (maybe you will need to
+refresh your page) :
+
+![](/public/screenshots/2020-05-26-choosenewtheme.png)
+
+We created our theme, and moved on it. At the moment, our new theme is a vanilla copy of `high-contrast light` but it's
+a start and we can now play with it (and it will be reflected in our convos in almost real time).
+
+**It is very easy to develop a new theme because without even having to restart convos, your changes are immediately
+viewable !!!**
+
+If it is your first theme, I recommend a light theme (bg light and text dark), it is easier.
+
+
+## How do I choose the colors ?
+
+Honestly I first started by typing **"hacker theme green"** in [google image](https://www.google.com/search?q=hacker+theme+green&tbm=isch&ved=2ahUKEwjljK2Zwc_pAhUMHBoKHU3MDPQQ2-cCegQIABAA&oq=hacker+theme+green&gs_lcp=CgNpbWcQAzoCCCk6BAgAEBM6AggAOgQIABAeOgYIABAIEB5QnwlYhxVgmRdoAXAAeACAAWaIAfQEkgEDNS4ymAEAoAEBqgELZ3dzLXdpei1pbWc&sclient=img&ei=V_zLXqWFE4y4aM2Ys6AP&bih=771&biw=1533&client=ubuntu&hs=Vge&safe=active) and I looked at results.
+
+![](/public/screenshots/2020-05-26-google.png)
+
+It gave me an idea about what colors to put together : 
+* First the green is really flashy, let’s go for flashy then.
+* There is black background or eventually dark grey background (cool, I will try to put this grey on one panel, I
+  usually don't like having all convos panel the same color)
+* Text colors are usually green or grey
+
+To have the color of the grey used in one of the screenshots (the grey in the treeview renders well with overall black
```

Comment:
I'm not a huge fan of a lot of parenthesis, but it's also not a show stopper.

Reply:
I removed some of them